onnxscript/tests/models/pnp.py

```diff
@@ -0,0 +1,262 @@
+import numpy as np
+
+import onnx
+from onnx import TensorProto
+from onnx.helper import make_tensor
+
+from onnxscript import script, graph
+# from onnxscript.onnx_opset import opset18ext as op
+from onnxscript.onnx_types import FLOAT, INT64
+
+from typing import (
+    Any,
+    Callable,
+    Mapping,
+)
+
+from onnxscript.onnx_opset._impl.opset18 import Opset18
+from onnxscript import tensor as onnxscript_tensor
+from onnxscript.onnx_types import (
+    BFLOAT16,
+    BOOL,
+    COMPLEX64,
+    COMPLEX128,
+    DOUBLE,
+    FLOAT,
+    FLOAT16,
+    INT8,
+    INT16,
+    INT32,
+    INT64,
+    STRING,
+    UINT8,
+    UINT16,
+    UINT32,
+    UINT64,
+)
+from onnxscript.values import Op, Opset
+from typing import Callable, Optional, Sequence, Tuple, Union
+
+import numpy as np
+import onnxruntime as rt
+import onnx
+
+class Opset18Ext(Opset18):
+    def __new__(cls):
+        return Opset.__new__(cls, "", 18)
+
+    def __init__(self):
+        super().__init__()
+
+    def OpaqueOp(self, input: FLOAT, model_path: STRING,
+    ) -> FLOAT:
+        model = onnx.load(model_path)
+        onnx.checker.check_model(model)
+        sess = rt.InferenceSession(model.SerializeToString(), providers=["CPUExecutionProvider"])
+
+        def op(input):
+            if isinstance(input, np.ndarray):
+                pred_roi = sess.run(None, {"image_roi": input})
+            else:
+                pred_roi = sess.run(None, {"image_roi": input.value})
+            pred_roi_tensor = onnxscript_tensor.Tensor(pred_roi[0])
+            # pred_roi_tensor = onnxscript_tensor.Tensor(input.value + np.ones(input.shape, dtype=input.dtype))
+            if isinstance(input, np.ndarray):
+                return pred_roi_tensor.value
+            else:
+                return pred_roi_tensor
+
+        return op(input)
+
+op = Opset18Ext()
+
+@script()
+def extend_roi_indices(roi_indices: INT64["roi_D", "roi_H", "roi_W", 3], seg_C: INT64) -> INT64[1, "seg_C", "roi_D", "roi_H", "roi_W", 5]:
+    """
+    This function is used to extend the roi indices to the segment dimension.
+    """
+    zero = op.Constant(value_int=0)
+    one = op.Constant(value_int=1)
+    one_ = op.Constant(value_ints=[1])
+    roi_D, roi_H, roi_W, _ = op.Split(op.Shape(roi_indices), num_outputs=4)
+    shape_1_seg_C_roi_D_H_W_1 = op.Concat(one_, seg_C, roi_D, roi_H, roi_W, one_, axis=0)

```

Comment:
nit: you can directly use `op.Shape(roi_indices)` instead of `roi_D, roi_H, roi_W` ... no need to split it and concat it back.